tests/test_analysis.py

```diff
@@ -1412,3 +1413,94 @@ def test_analysis_upload_file_makes_expected_calls_s3(
         [mock.call({"upload_type": "s3", "other_info": "test_value"}, "test_tag", fp)]
     )
     analysis._client._azure_upload.assert_has_calls([])
+
+
+def test_is_retraining_not_retraining(analysis_data, patched_fero_client):
+    """Test that is_retraining() returns false if the analysis is not training"""
+
+    patched_fero_client.get.return_value = analysis_data.copy()
+    analysis_data["latest_revision_model_state"] = "T"
```

Comment:
Shouldn't this status be an `"I"` or something other than training based on the description of the test?

Reply:
We set that to make sure it's actually reloading based on the latest status of analysis, not the current cached value.

Basically, it covers the path like this.
```python
analysis.revise()  # internal data is 'T'
analysis.is_retraining()  # would still be training if we don't refresh before checking
```